src/main/java/ifsul/agileproject/rachadinha/domain/entity/User.java

```diff
@@ -0,0 +1,42 @@
+//Classe Usuário
+package ifsul.agileproject.rachadinha.domain.entity;
+
+import jakarta.persistence.*;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.ArrayList;
+
+@Entity
+@AllArgsConstructor
+@NoArgsConstructor
+@Data
+@Table(name = "Usuario")
+public class User {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id")
+    private int id;
+
+    @Column(name = "name")
+    private String name;
+
+    @Column(name = "email", unique = true)
+    private String email;
+
+    @Column(name = "password")
+    private String password;
+
+    //@ManyToMany
+    //ArrayList<User> listaAmigos;
+
+    public User(String name, String email, String password) {
```

Comment:
Couldn't we use lombok to auto generate a constructor like this one?

Reply:
Lombok generates constructors that take no arguments, one argument per final / non-null field, or one argument for every field. So, for a constructor like this one, lombok don't help (in this case)

Reply:
If you prefer, Lombok has the @RequiredArgsConstructor annotation,
that all fields marked with @NonNull are included in the custom constructor that @edukatiau made.

Ref: https://projectlombok.org/features/constructor